docs/language/src/builtins/builtin_elements.md

```diff
@@ -442,340 +446,309 @@ location specified by the `x` and `y` properties, using the control points speci
--   **`font-weight`** (_int_): The weight of the font. The values range from 100 (lightest) to 900 (thickest). 400 is the normal weight.
--   **`color`** (_brush_): The color of the text (default value: depends on the style)
--   **`horizontal-alignment`** (_enum [`TextHorizontalAlignment`](builtin_enums.md#texthorizontalalignment)_): The horizontal alignment of the text.
--   **`vertical-alignment`** (_enum [`TextVerticalAlignment`](builtin_enums.md#textverticalalignment)_): The vertical alignment of the text.
--   **`has-focus`** (_bool_): Set to `true` when item is focused and receives keyboard events.
--   **`letter-spacing`** (_length_): The letter spacing allows changing the spacing between the glyphs. A positive value increases the spacing
-    and a negative value decreases the distance. The default value is 0.
--   **`single-line`** (_bool_): When set to `true`, no newlines are allowed (default value: `true`)
--   **`read-only`** (_bool_): When set to `true`, text editing via keyboard and mouse is disabled but
-    selecting text is still enabled as well as editing text programatically (default value: `false`)
--   **`wrap`** (_enum [`TextWrap`](builtin_enums.md#textwrap)_): The way the text input wraps. Only makes sense when `single-line` is false. (default: no-wrap)
--   **`input-type`** (_enum [`InputType`](builtin_enums.md#inputtype)_): The way to allow special input viewing properties such as password fields (default value: `text`).
-
-### Methods
-
--   **`focus()`** Call this function to focus the text input and make it receive future keyboard events.
-
-### Callbacks
-
--   **`accepted()`**: Emitted when enter key is pressed
--   **`edited()`**: Emitted when the text has changed because the user modified it
--   **`cursor-position-changed(Point)`**: The cursor was moved to the new (x, y) position
-
-### Example
+This example breaks a longer paragraph of text into multiple lines, by setting a `wrap`
+policy and assigning a limited `width` and enough `height` for the text to flow down:
 
 ```slint
 export component Example inherits Window {
     width: 270px;
-    height: 100px;
+    height: 300px;
 
-    TextInput {
-        text: "Replace me with a name";
+    Text {
+        x:0;
+        text: "This paragraph breaks into multiple lines of text";
+        wrap: word-wrap;
+        width: 150px;
+        height: 100%;
     }
 }
 ```
 
-## `PopupWindow`
+## `TouchArea`
 
-This allow to show a popup window like a tooltip or a popup menu.
+Use `TouchArea` to control what happens when the it covers is touched or interacted with
+using the mouse.
 
-Note: It is not allowed to access properties of elements within the popup from outside of the popup.
+When not part of a layout, its width or height default to 100% of the parent element.
 
-### Methods
+### Properties
 
--   **`show()`** Call this function to show the popup.
+-   **`has-hover`** (_out_ _bool_): Set to `true` when the mouse is over the `TouchArea`.
```

Comment:
Same as in `TextInput`, emphasis that it's not for the user to set (they'll also get an error, of course).

```suggestion
-   **`has-hover`** (_out_ _bool_): `TouchArea` sets this to `true` when the mouse is over it.
```